Lib/glyphsLib/builder/components.py

```diff
@@ -51,6 +69,50 @@ def to_ufo_components(self, ufo_glyph, layer):
             ufo_glyph.lib[_lib_key(key)] = values
 
 
+def to_ufo_components_decompose(self, ufo_glyph, layer):
+    """Draw decomposed .glyphs components with a pen, adding them to the parent
+    glyph."""
+
+    layer_id = layer.foreground.layerId
+    layer_master_id = layer.foreground.associatedMasterId
+
+    if layer_id in self._glyph_sets:
+        layers = self._glyph_sets[layer_id]
+    else:
+        if layer_id == layer_master_id:
+            # Is a master layer.
+            layers = self._glyph_sets[layer_id] = {
+                g.name: l
+                for g in layer.parent.parent.glyphs
+                for l in g.layers
+                if l.layerId == layer_id
+            }
+        else:
+            # Is a non-master layer.
+            layers_nonmaster = {
+                g.name: l
+                for g in layer.parent.parent.glyphs
+                for l in g.layers
+                if l.name == layer.name
+            }
+            # FIXME: Have to use layer ID instead of name because glyphsLib collates
+            # layers with the same name.
```

Comment:
WDYM by "glyphsLib collates layers with the same name"?

Reply:
If you have a regular layer in Glyphs and duplicate it twice, `len(Font.glyphs["A"].layers) == 3` but it's `1` in glyphsLib. Some layer wins I guess. This is significant because layers are looked up by names and the first one wins, whcih would be different in glyphsLib.

Reply:
Oh, it was a source error.